assets/javascripts/lib/uploads.js

```diff
@@ -47,3 +54,37 @@ export function generateUploadKey() {
       .then(resolve, reject);
   });
 }
+
+export function downloadEncryptedFile(url, keyPromise, opts) {
+  opts = opts || {};
+
+  const downloadPromise = new Promise((resolve, reject) => {
+    let req = new XMLHttpRequest();
+    req.open("GET", url, true);
+    req.responseType = "arraybuffer";
+    req.onload = function () {
+      let filename = req.getResponseHeader("Content-Disposition");
+      if (filename) {
+        // Requires Access-Control-Expose-Headers: Content-Disposition.
+        filename = filename.match(/filename="(.*?)"/)[1];
+      }
+      resolve({ buffer: req.response, filename });
+    };
+    req.onerror = reject;
+    req.send(null);
```

Comment:
```suggestion
    req.send();
```
This should be the same as `null` right?

Reply:
Yes, you are right. MDN says the same.